### PR TITLE
Fix ambiguous segfault for test requiring MORPHEUS_ROOT

### LINKED ISSUE #401: [BUG] C++ tests fail due to segfault
**Describe the bug**
Morpheus 22.11 tests fail with segfault.

**Steps/Code to reproduce bug**
Build morpheus dev container according to README/CONTRIBUTING guides. Launch dev container. Build morpheus with tests enabled (`-DMORPHEUS_BUILD_TESTS=ON`). Run tests:

**Expected behavior**
Tests run to completion and pass.

**Actual behavior**
Segmentation fault on tests `TestMatxUtil.ReduceMax2dColMajor` and `TestMasking.Ranges`:
```
(morpheus) root@charris-dt:/workspace# ./build/morpheus/_lib/tests/test_libmorpheus.x --gtest_filter=TestMatxUtil.ReduceMax2dColMajor
Note: Google Test filter = TestMatxUtil.ReduceMax2dColMajor
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestMatxUtil
[ RUN      ] TestMatxUtil.ReduceMax2dColMajor
[charris-dt:4466 :0:4466] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace (tid:   4466) ====
 0  /opt/conda/envs/morpheus/lib/./libucs.so.0(ucs_handle_error+0x155) [0x7f787de7d3f5]
 1  /opt/conda/envs/morpheus/lib/./libucs.so.0(+0x2d791) [0x7f787de7d791]
 2  /opt/conda/envs/morpheus/lib/./libucs.so.0(+0x2d962) [0x7f787de7d962]
 3  /usr/lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7f78881b4090]
 4  /usr/lib/x86_64-linux-gnu/libc.so.6(+0x1886e5) [0x7f78882f96e5]
 5  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x2a28c) [0x56342ecb528c]
 6  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x4e) [0x7f789ffff98e]
 7  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing4Test3RunEv+0xe1) [0x7f789ffffbe1]
 8  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8TestInfo3RunEv+0x13f) [0x7f789fffff0f]
 9  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing9TestSuite3RunEv+0x106) [0x7f78a0000036]
10  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8internal12UnitTestImpl11RunAllTestsEv+0x4dc) [0x7f78a00005ec]
11  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8UnitTest3RunEv+0xd9) [0x7f78a0000859]
12  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x17ae3) [0x56342eca2ae3]
13  /usr/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7f7888195083]
14  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x18fbd) [0x56342eca3fbd]
=================================
Segmentation fault (core dumped)
(morpheus) root@charris-dt:/workspace# ./build/morpheus/_lib/tests/test_libmorpheus.x --gtest_filter=TestMasking.Ranges              
Note: Google Test filter = TestMasking.Ranges
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from TestMasking
[ RUN      ] TestMasking.Ranges
[charris-dt:4468 :0:4468] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
==== backtrace (tid:   4468) ====
 0  /opt/conda/envs/morpheus/lib/./libucs.so.0(ucs_handle_error+0x155) [0x7fde0a70b3f5]
 1  /opt/conda/envs/morpheus/lib/./libucs.so.0(+0x2d791) [0x7fde0a70b791]
 2  /opt/conda/envs/morpheus/lib/./libucs.so.0(+0x2d962) [0x7fde0a70b962]
 3  /usr/lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7fde14a42090]
 4  /usr/lib/x86_64-linux-gnu/libc.so.6(+0x1886e5) [0x7fde14b876e5]
 5  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x4390d) [0x559f477c390d]
 6  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x4e) [0x7fde2c88d98e]
 7  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing4Test3RunEv+0xe1) [0x7fde2c88dbe1]
 8  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8TestInfo3RunEv+0x13f) [0x7fde2c88df0f]
 9  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing9TestSuite3RunEv+0x106) [0x7fde2c88e036]
10  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8internal12UnitTestImpl11RunAllTestsEv+0x4dc) [0x7fde2c88e5ec]
11  /opt/conda/envs/morpheus/lib/libgtest.so(_ZN7testing8UnitTest3RunEv+0xd9) [0x7fde2c88e859]
12  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x17ae3) [0x559f47797ae3]
13  /usr/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fde14a23083]
14  ./build/morpheus/_lib/tests/test_libmorpheus.x(+0x18fbd) [0x559f47798fbd]
=================================
Segmentation fault (core dumped)
```

**Additional context**
- branch-22.11

### PR DESCRIPTION
**Pull request by cwharris**
Fixes #401

### CODE REVIEW

**Comment by cwharris** [comment]
rerun tests

File: morpheus/_lib/tests/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ list(APPEND CMAKE_MESSAGE_CONTEXT "tests")
 # Keep all source files sorted
 add_executable(test_libmorpheus
   # test_cuda.cu
+  test_morpheus.cpp
```
Comment (mdemoret-nv) [inline morpheus/_lib/tests/CMakeLists.txt:21]: Keep these sorted.

File: morpheus/_lib/tests/test_morpheus.hpp [hunk 2]
```diff
@@ -20,6 +20,12 @@
...
 #include <gtest/gtest.h>   // IWYU pragma: keep
 
+#include <filesystem>
+
 #define TEST_CLASS(name)                      \
     class Test##name : public ::testing::Test \
     {}
+
+namespace morpheus::test {
+std::filesystem::path get_morpheus_root();
```
Comment (mdemoret-nv) [inline morpheus/_lib/tests/test_morpheus.hpp:30]: Add a doxygen string here describing the method

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]

**Review by mdemoret-nv — APPROVED** [review]
Can merge once CI is passing

**Comment by cwharris** [comment]
@gpucibot merge

**Pull request merged**